src/passes/externalContractHandler/externalContractInterfaceInserter.ts

```diff
@@ -0,0 +1,154 @@
+import assert = require('assert');
+import {
+  ContractDefinition,
+  ContractKind,
+  FunctionCall,
+  FunctionDefinition,
+  FunctionKind,
+  getNodeType,
+  Identifier,
+  MemberAccess,
+  SourceUnit,
+  UserDefinedType,
+  UserDefinedTypeName,
+  VariableDeclaration,
+} from 'solc-typed-ast';
+
+import { AST } from '../../ast/ast';
+import { CairoContract } from '../../ast/cairoNodes';
+import { ASTMapper } from '../../ast/mapper';
+import { cloneASTNode } from '../../utils/cloning';
+
+function genContractInterface(
+  contract: ContractDefinition,
+  sourceUnit: SourceUnit,
+  ast: AST,
+): ContractDefinition {
+  const contractId = ast.reserveId();
+  const contractInterface = new CairoContract(
+    contractId,
+    '',
+    `${contract.name}@interface`,
+    sourceUnit.id,
+    ContractKind.Interface,
+    contract.abstract,
+    false,
+    contract.linearizedBaseContracts,
+    contract.usedErrors,
+    new Map(),
+    0,
+  );
+
+  contract.vFunctions.forEach((func) => {
+    // return if func is constructor
+    if (func.kind === FunctionKind.Constructor) return;
+
+    const funcClone = new FunctionDefinition(
+      ast.reserveId(),
+      '',
+      contractInterface.id,
+      func.kind,
+      func.name,
+      func.virtual,
+      func.visibility,
+      func.stateMutability,
+      func.isConstructor,
+      cloneASTNode(func.vParameters, ast),
+      cloneASTNode(func.vReturnParameters, ast),
+      func.vModifiers,
```

Comment:
Modifiers would also need to be cloned too. I would suggest doing as is done in `ExternalFunctionCreator.createExternalFunctionDefinition` (and ideally then refactoring the two into a `cloneFunctionWithoutBody` in `cloning.ts`) where the method is to disconnect the body, clone the FunctionDefinition, then reconnect the body to the original FunctionDefinition

Reply:
Done!